esp-hal/src/dma/mod.rs

```diff
@@ -1307,8 +1434,10 @@ where
     C: ChannelTypes,
     MODE: Mode,
 {
+    /// TX half of the channel
```

Comment:
I really don't like it but the LCD-i8080 driver currently wants only the TX halve 🤔

Reply:
I'm gonna be taking the RX half in the camera side soon too :sweat_smile:.

I've been wondering though, might it make sense for the `Dma` struct to have a channel creator for each half instead of the whole channel?
Besides async memcpy there's no reason to keep the rx and tx sides tied together.

Reply:
The original reason to not expose the halves initially was that on ESP32/ESP32-S2 the channels are dedicated to their peripherals and are not freely useable.

I think we should revisit that in another PR (mostly find a nicer way ... e.g. separate channel creators or something - actually what I wanted to think about during the weekend but the PR as is shouldn't make anything worse so I want to tackle that later)